dotnet/src/Microsoft.Agents.AI.Hosting.OpenAI/Responses/HostedAgentResponseExecutor.cs

```diff
@@ -93,33 +127,18 @@ private AIAgent ResolveAgent(CreateResponse request)
             agentName = entityId;
         }
 
-        // Never use model field for entity ID - it's for the actual model name
-        if (string.IsNullOrEmpty(agentName))
-        {
-            throw new InvalidOperationException("No 'agent.name' or 'metadata.entity_id' specified in the request.");
-        }
-
-        // Resolve the keyed agent service
-        try
-        {
-            return this._serviceProvider.GetRequiredKeyedService<AIAgent>(agentName);
-        }
-        catch (InvalidOperationException ex)
-        {
-            this._logger.LogError(ex, "Failed to resolve agent with name '{AgentName}'", agentName);
-            throw new InvalidOperationException($"Agent '{agentName}' not found. Ensure the agent is registered with AddAIAgent().", ex);
-        }
+        return agentName;
     }
 
     /// <summary>
-    /// Validates that the agent can be resolved without actually resolving it.
-    /// This allows early validation before starting async execution.
+    /// Extracts the model ID from the request. Returns null if model is being used as agent name.
     /// </summary>
     /// <param name="request">The create response request.</param>
-    /// <exception cref="InvalidOperationException">Thrown when the agent cannot be resolved.</exception>
-    public void ValidateAgent(CreateResponse request)
+    /// <returns>The model ID to use, or null if model parameter is being used for agent identification.</returns>
+    private static string? GetModelId(CreateResponse request)
     {
-        // Use the same logic as ResolveAgent but don't return the agent
-        _ = this.ResolveAgent(request);
+        // If agent.name is specified, use model as the model ID
+        // Otherwise, model is being used for agent name, so return null
+        return request.Agent?.Name is not null ? request.Model : null;
```

Comment:
Is this a temporary thing?

Reply:
oops, yep, bad rebase. Thanks for catching